docs/rfcs/xxxx-integrations-in-operator.md

```diff
@@ -0,0 +1,234 @@
+* `__meta_agentoperator_integration_cr_label_<labelname>`: Each label from the MetricsIntegration CR
+* `__meta_agentoperator_integration_cr_labelpresent_<labelname>`: `true` for each label from the MetricsIntegration CR
+
+### Discovering running integrations
+
+[#1198][] introduces a new service discovery (SD) API to retrieve a list of
+running integrations. The endpoint is available at
+`/agent/api/v1/metrics/integrations/sd` and returns JSON which conforms to
+Prometheus' [`http_sd_config`][http_sd_config] API.
+
+Note that integration targets returned by this API always have the following
+base labels:
+
+* `instance`: The inferred `instance` key based on integration settigs.
+* `job`: `integrations/<integration name>`
+* `agent_hostname`: Hostname of the agent running the integration. This will be
+  the Node name for `daemonset` integrations, otherwise will be the Pod name.
+* `__meta_agent_integration_name`: Name of the integration, e.g., `node_exporter`
+* `__meta_agent_integration_instance`: The inferred `instance` key based on
+  integration settings. Same as `instance`.
+* `__meta_agent_integration_autoscrape`: `true` or `false` depending on whether
+  autoscrape is enabled.
+
+These labels are available on top of any custom labels supplied in the `labels`
+field.
+
+Pods created for MetricsInstances will use `http_sd_config` to find, filter,
+and scrape metrics for integrations. However, there is a problem:
+`http_sd_config` can only be configured with one known URL, but there may be
+any number of pods which are running integrations.
+
+The operator's existing HTTP server will be extended with a new endpoint which
+aggregates the integrations targets for all managed integrations pods. This
+endpoint will be available at `/operator/api/v1/metrics/integrations/sd`. For
+this endpoint to work, the Operator will create a new controller that watches
+for changes to Endpoints related to the integrations-specific Services created
+by the operator. The list of relevant pod IPs will then be maintained
+internally.
+
+When the operator's integration SD endpoint is invoked, it will parallelize
+over all pods, invoking their individual SD API endpoints, and aggregating the
+responses into a final set. Failed API requests to pods will be logged but not
+fail the operator's request.
+
+Finally, on top of the target's labels, two additional labels will be added:
+
+* `__meta_agentoperator_grafanaagent_name`: Name of the owning GrafanaAgent.
+* `__meta_agentoperator_grafanaagent_namespace`: Namespace of the owning GrafanaAgent.
+
+As a result, there is a centralized API to discover metrics endpoints for all
+running integrations, including information about the integration, the owning
+MetricsIntegration CR, and the root GrafanaAgent resource.
+
+### Scraping integrations
+
+The other new CRD, IntegrationMonitor, will be used to scrape running
+integrations. IntegrationsMonitors are discovered by a MetricsInstance, and
```

Comment:
```suggestion
integrations. IntegrationMonitor's are discovered by a MetricsInstance, and
```

Reply:
The apostrophe isn't correct here :)

Reply:
intended just to switch IntegrationsMonitor to IntegrationMonitor

Reply:
ah, didn't see that. fixed, thanks!